src/client/QXmppCallInviteManager.h

```diff
@@ -0,0 +1,123 @@
+// SPDX-FileCopyrightText: 2023 Tibor Csötönyi <work@taibsu.de>
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#ifndef QXMPPCALLINVITEMANAGER_H
+#define QXMPPCALLINVITEMANAGER_H
+
+#include "QXmppClientExtension.h"
+#include "QXmppError.h"
+#include "QXmppJingleIq.h"
+#include "QXmppMessageHandler.h"
+#include "QXmppSendResult.h"
+#include "QXmppTask.h"
+
+class QXmppCallInviteManager;
+class QXmppCallInvitePrivate;
+class QXmppCallInviteManagerPrivate;
+
+class QXMPP_EXPORT QXmppCallInvite : public QObject
+{
+    Q_OBJECT
+public:
+    struct Rejected
+    {
+    };
+    struct Retracted
+    {
+    };
+    struct Left
+    {
+    };
+
+    using Result = std::variant<Rejected, Retracted, Left, QXmppError>;
+
+    QXmppCallInvite(QXmppCallInviteManager *manager);
```

Comment:
should be `explicit` to prevent implicit conversion of CallInviteManager* into CallInvite.